Fabric.Authorization.UnitTests/Users/UsersModuleTests.cs

```diff
@@ -99,6 +99,106 @@ public void GetUserPermissions_NoParameters_DefaultsToTopLevel()
             AssertOk(result, 3);
         }
 
+        [Theory]
+        [InlineData("test", "", "You must specify an IdentityProvider for this user")]
+        [InlineData("", "test", "You must specify a SubjectId for this user")]
+        public void AddUser_MissingData_ReturnsBadRequest(string subjectId, string identityProvider, string message)
+        {
+            var existingClient = ExistingClients.First();
+            var usersModule = CreateBrowser(
+                new Claim(Claims.Scope, Scopes.WriteScope),
+                new Claim(Claims.ClientId, existingClient.Id)
+            );
+
+            var postResponse = usersModule.Post("/user", with =>
+            {
+                with.HttpRequest();
+                with.JsonBody(new
+                {
+                    SubjectId = subjectId,
+                    IdentityProvider = identityProvider
+                });
+            }).Result;
```

Comment:
is it ok to use .Result here?

Reply:
The unit test use Mock stores which aren't async, so we aren't getting deadlocks here. We should convert them to async, just haven't yet.